pkg/controller/bypass/bypass_test.go

```diff
@@ -25,24 +25,39 @@ import (
 
 	"github.com/agiledragon/gomonkey/v2"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 	"istio.io/api/annotation"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
 )
 
-func TestPodSidecarLabelChangeTriggersAddIptablesAction(t *testing.T) {
-	client := fake.NewSimpleClientset()
-
+func TestBypassController(t *testing.T) {
 	err := os.Setenv("NODE_NAME", "test_node")
-	require.NoError(t, err)
+	assert.NoError(t, err)
 	t.Cleanup(func() {
 		os.Unsetenv("NODE_NAME")
 	})
-	stopCh := make(<-chan struct{})
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	namespaceName := "default"
+	namespace := &corev1.Namespace{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: namespaceName,
+			Labels: map[string]string{
+				"istio-injection": "enabled",
+			},
+		},
+	}
+	client := fake.NewSimpleClientset(namespace)
 	c := NewByPassController(client)
 	go c.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, c.pod.HasSynced)
```

Comment:
This is the root cause

Reply:
you can directly call `c.Run(stopCh)`, just remove `go`

Reply:
Yes,  here we can